.editorconfig

```
@@ -0,0 +1,18 @@
+# https://EditorConfig.org
+
+# top-most EditorConfig file
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
+indent_size = 2
+indent_style = tab
+insert_final_newline = true
+trim_trailing_whitespace = false
+
+[{*.js,*.ts,}]
+indent_size = 4
+
+[{.*,*.{md|mdx},*.json,*.toml,*.yml,}]
```

```suggestion
[{.*,*.md,*.mdx,*.json,*.toml,*.yaml,*.yml,}]
indent_size = 2
```

Two changes here:
- The nested alternatives {md|mdx} don't seem to work properly for me in VS Code's EditorConfig extension, so I expanded this to two options. I also added `yaml` due to pnpm using it.
- As these extensions are actually using spaces, this is where we should define the `indent_size`, so I moved it here.